.github/workflows/ci-cd.yaml

```diff
@@ -0,0 +1,38 @@
+name: CI/CD
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+  repository_dispatch:
+    types: [remote-build]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+          ref: ${{ github.head_ref }}
+      - name: Update Submodule
+        run: |
+          git submodule update --init --force --recursive --remote
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: npm install
```

Comment:
Don't really know why, but npm ci fails here (it complains about package-lock.json), even though it works locally.

Reply:
make sure to commit your package-lock.json
then you could try `npm ci` locally and here (in the CI)
it will use the version form package-lock.json

Reply:
you can 'thank' the luck. IMHO it's how it works in the node world :p